benchmark/BenchmarkMain.cpp

```diff
@@ -0,0 +1,221 @@
+      // Row name
+      (*stringStream) << "\n";
+      addStringWithPadding(stringStream, table.rowNames[row],
+          rowNameMaxStringWidth);
+      
+      // Row content.
+      for (size_t column = 0; column < numberColumns; column++) {
+        (*stringStream) << columnSeperator;
+        addStringWithPadding(stringStream,
+            optionalFloatToString(table.entries[row][column]),
+            columnMaxStringWidth[column]);
+      }
+    }
+  }
+}
+
+}
+
+/*
+ * @brief Goes through all types of registered benchmarks, measures their time
+ * and prints their measured time in a fitting format.
+ */
+int main() {
+  // Measuering the time for all registered benchmarks.
+  // For measuring and saving the times.
+  BenchmarkRecords records;
+ 
+  // Goes through all registered benchmarks and measures them.
+  for (const auto& benchmarkFunction: BenchmarkRegister::getRegisteredBenchmarks()) {
+    benchmarkFunction(&records);
+  }
+
+  // The values for all the categories of benchmarks.
+  const std::vector<BenchmarkRecords::RecordEntry>& singleMeasurements =
+    records.getSingleMeasurements();
+  const std::vector<BenchmarkRecords::RecordGroup>& recordGroups =
+    records.getGroups();
+  const std::vector<BenchmarkRecords::RecordTable>& recordTables =
+    records.getTables();
+
+  // Visualizes the measured times.
+  std::stringstream visualization;
+  
+  // @brief Adds a category to the string steam, if it is not empty. Mainly
+  //  exists for reducing code duplication.
+  //
+  // @param stringStream The stringstream where the text will get added.
+  // @param categoryRecord The information needed, for printing the benchmark
+  //  category. Should be a vector of RecordEntry, RecordGroup, or RecordTable.
+  // @param categoryAddPrintStreamFunction The function, which add the
+  //  benchmark category information, convert them to text, and add that text
+  //  to the stringstream.
+  // @param suffix Added to the stringstream after
+  //  categoryAddPrintStreamFunction. Mostly for linebreaks between those
+  //  categories.
+  auto addNonEmptyCategorieToStringSteam = [](std::stringstream* stringStream,
+      const auto& categoryRecord, const auto& categoryAddPrintStreamFunction,
+      const std::string& suffix = ""){
+    if (categoryRecord.size() > 0){
+      categoryAddPrintStreamFunction(stringStream, categoryRecord);
+      (*stringStream) << suffix;
+    }
+  };
+
+  // Visualization for single measurments, if there are any.
+  addNonEmptyCategorieToStringSteam(&visualization, singleMeasurements,
+      addSingleMeasurementsToStringstream, "\n\n");
+
+  // Visualization for groups, if there are any.
+  addNonEmptyCategorieToStringSteam(&visualization, recordGroups,
+      addGroupsToStringstream, "\n\n");
+
+  // Visualization for tables, if there are any.
+  addNonEmptyCategorieToStringSteam(&visualization, recordTables,
+      addTablesToStringstream);
+
+  std::cout << visualization.str() << "\n";
+};
```

Comment:
Maybe Move the complete functionality of this file as is to a class/ pair of .h .cpp file
`BenchmarkToString` or `BenchmarkToConsoleOutput` etc, then the main function will become very very short and it will be easier to add completely different outputs to the main function.

Reply:
Done, see [b2d29e6](https://github.com/ad-freiburg/qlever/pull/860/commits/b2d29e61e7f29808fa61b79ba10c6eb719252ee4).

I also moved the conversion to json into it's own file. Is that alright, @joka921 ?